File: universal-utils/src/main/scala/io/github/ppdzm/utils/universal/base/Constructor.scala
package io.github.ppdzm.utils.universal.base

import java.sql.Date

object Constructor {

    def construct(obj: AnyRef, fieldsNameValuePair: Map[String, Any]): Unit = {
        obj.getClass.getDeclaredFields.foreach {
            field =>
                val columnName = field.getName
                if (fieldsNameValuePair.contains(columnName) && fieldsNameValuePair(columnName) != null) {
                    field.setAccessible(true)
                    field.getType.getSimpleName.toLowerCase match {
                        case "integer" => field.set(obj, fieldsNameValuePair.get(columnName).asInstanceOf[Int])
                        case "long" => field.set(obj, fieldsNameValuePair.get(columnName).asInstanceOf[Long])
                        case "date" => field.set(obj, Date.valueOf(fieldsNameValuePair.get(columnName).toString))
                        case _ => field.set(obj, fieldsNameValuePair.get(columnName))
                    }
                }
        }
    }

}

File: universal-utils/src/main/java/io/github/ppdzm/utils/universal/feature/Constructor.java
package io.github.ppdzm.utils.universal.feature;

import java.util.Arrays;
import java.util.Map;

/**
 * @author Created by Stuart Alex on 2023/5/13.
 */
public class Constructor {

    public static <T> T construct(Class<T> tClass, Map<String, Object> fieldValueMap) throws InstantiationException, IllegalAccessException {
        T t = tClass.newInstance();
        construct(t, fieldValueMap);
        return t;
    }

    public static void construct(Object object, Map<String, Object> fieldValueMap) {
        Arrays.stream(object.getClass().getDeclaredFields()).forEach(field -> {
            String columnName = field.getName();
            try {
                if (fieldValueMap.containsKey(columnName) && fieldValueMap.get(columnName) != null) {
                    field.setAccessible(true);
                    field.set(object, fieldValueMap.get(columnName));
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
        });
    }

}